jetty-http/src/main/java/org/eclipse/jetty/http/GZIPContentDecoder.java

```diff
@@ -209,6 +221,13 @@ else if ((_flags & 0x2) == 0x2)
                             if (buffer == null)
                                 buffer = acquire(_bufferSize);
 
+                            if (_inflater.needsInput())
+                            {
+                                if (!compressed.hasRemaining())
+                                    return;
+                                _inflater.setInput(compressed);
+                            }
+
```

Comment:
(I moved this up because otherwise there is a call done to _inflater.inflate with no input set yet)